# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasUtils.cpp
# Commit: ff14d3c64e53
# Full Hash: ff14d3c64e53dad7b4720c983b8f19bb494ed28a
# Author: Ehsan Akhgari <ehsan@mozilla.com>
# Date: 2019-03-06 04:36:55
# Regressor Bug: 1532414
# File Overlap Count: 1
# Description:
#   Bug 1532414 - Pass a document/principal pointer to nsContentUtils::ShouldResistFingerprinting() callers in the canvas API; r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22002
# ==============================================================================

diff -r 14b6b6f8a599 -r ff14d3c64e53 dom/canvas/CanvasUtils.cpp
--- a/dom/canvas/CanvasUtils.cpp	Tue Mar 05 12:42:51 2019 +0000
+++ b/dom/canvas/CanvasUtils.cpp	Tue Mar 05 12:42:51 2019 +0000
@@ -49,7 +49,7 @@
 bool IsImageExtractionAllowed(Document* aDocument, JSContext* aCx,
                               nsIPrincipal& aPrincipal) {
   // Do the rest of the checks only if privacy.resistFingerprinting is on.
-  if (!nsContentUtils::ShouldResistFingerprinting()) {
+  if (!nsContentUtils::ShouldResistFingerprinting(aDocument)) {
     return true;
   }
 